### PR TITLE
Added checking contributors from friendlist when creating and editing…

### PR DESCRIPTION
**Pull request by niklaus64**
… project

### CODE REVIEW

**Review by GitSakro — DISMISSED** [review]
Delete Incjection SelectedProject from ANYWHERE

**Comment by niklaus64** [comment]
Of course I will fix your mistakes

**Comment by mdjdrn1** [comment]
DO NOT MERGE, SOMETHING IS BROKEN

**Review by mdjdrn1 — APPROVED** [review]
Now it's working

File: src/main/java/textEditor/controller/AddProjectController.java [hunk 1]
```diff
@@ -13,16 +16,19 @@
...
 import java.rmi.RemoteException;
-import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.ResourceBundle;
 
 public class AddProjectController implements Initializable, ClientInjectionTarget, WindowSwitcherInjectionTarget, UserInjectionTarget {
     @FXML
-    public TextField projectNameField, contributorsField;
+    public TextField projectNameField;
```
Comment (mdjdrn1) [inline src/main/java/textEditor/controller/AddProjectController.java]: Change access to private

File: src/main/java/textEditor/controller/AddProjectController.java [hunk 2]
```diff
@@ -13,16 +16,19 @@
...
+    @FXML
+    public CheckComboBox contributorsField;
```
Comment (mdjdrn1) [inline src/main/java/textEditor/controller/AddProjectController.java]: Change access to private

File: src/main/java/textEditor/controller/AddProjectController.java [hunk 3]
```diff
@@ -13,16 +16,19 @@
...
     @FXML
     public TextArea projectDescriptionField;
+    @FXML
+    public Label information;
```
Comment (mdjdrn1) [inline src/main/java/textEditor/controller/AddProjectController.java]: Change access to private

File: src/main/java/textEditor/controller/AddProjectController.java [hunk 4]
```diff
@@ -13,16 +16,19 @@
...
 import java.util.ResourceBundle;
 
 public class AddProjectController implements Initializable, ClientInjectionTarget, WindowSwitcherInjectionTarget, UserInjectionTarget {
     @FXML
-    public TextField projectNameField, contributorsField;
+    public TextField projectNameField;
+    @FXML
+    public CheckComboBox contributorsField;
     @FXML
     public TextArea projectDescriptionField;
```
Comment (mdjdrn1) [inline src/main/java/textEditor/controller/AddProjectController.java]: Change access to private

File: src/main/java/textEditor/controller/AddProjectController.java [hunk 5]
```diff
@@ -59,15 +67,32 @@ public void injectWindowSwitcher(WindowSwitcher switcher) {
 
     public void addButtonClicked(ActionEvent actionEvent) {
         try {
-            List<String> arrayList = Arrays.asList(contributorsField.getText().split(","));
-            Project project = new ProjectImpl(0, projectNameField.getText(), projectDescriptionField.getText(), Arrays.asList(contributorsField.getText().split(",")));
+            List contributors = Arrays.asList(contributorsField.getCheckModel().getCheckedItems().toArray());
+            if(!contributors.contains(user))
```
Comment (mdjdrn1) [inline src/main/java/textEditor/controller/AddProjectController.java]: Reformat

File: src/main/java/textEditor/controller/AddProjectController.java [hunk 6]
```diff
@@ -59,15 +67,32 @@ public void injectWindowSwitcher(WindowSwitcher switcher) {
...
+            }
+            Project project = new ProjectImpl(0, projectNameField.getText(), projectDescriptionField.getText(), contributors);
             databaseModel.addProject(project);
             windowSwitcher.loadWindow(WindowSwitcher.Window.PICK_PROJECT);
         } catch (IOException e) {
             e.printStackTrace();
         }
     }
 
+    private void setInformation(String text)
```
Comment (mdjdrn1) [inline src/main/java/textEditor/controller/AddProjectController.java]: Reformat

File: src/main/java/textEditor/controller/EditProjectController.java [hunk 7]
```diff
@@ -38,7 +43,18 @@ public void cancelClicked(ActionEvent actionEvent) {
 
     public void applyClicked(ActionEvent actionEvent) {
         try {
-            Project editedProject = new ProjectImpl(project.getId(), projectNameField.getText(), projectDescriptionField.getText(), Arrays.asList(contributorsField.getText().split(",")));
+            List contributors = Arrays.asList(contributorsField.getCheckModel().getCheckedItems().toArray());
```
Comment (mdjdrn1) [inline src/main/java/textEditor/controller/EditProjectController.java:46]: Reformat

File: src/main/java/textEditor/controller/EditProjectController.java [hunk 8]
```diff
@@ -51,26 +67,49 @@ public void applyClicked(ActionEvent actionEvent) {
         }
     }
 
+    private void setInformation(String text)
```
Comment (mdjdrn1) [inline src/main/java/textEditor/controller/EditProjectController.java]: Reformat

File: src/main/java/textEditor/controller/EditProjectController.java [hunk 9]
```diff
@@ -51,26 +67,49 @@ public void applyClicked(ActionEvent actionEvent) {
...
     public void initialize(URL location, ResourceBundle resources) {
         try {
             databaseModel = (DatabaseModel) rmiClient.getModel("DatabaseModel");
+            projectNameField.setText(project.getTitle());
+            projectDescriptionField.setText(project.getDescription());
+
+            //Add contributors and friends
+            contributorsField.getItems().addAll(project.getContributors());
+            databaseModel.getFriends(user).forEach(user -> {
+                if(!contributorsField.getItems().contains(user))
```
Comment (mdjdrn1) [inline src/main/java/textEditor/controller/EditProjectController.java]: Reformat

File: src/main/java/textEditor/controller/EditProjectController.java [hunk 10]
```diff
@@ -51,26 +67,49 @@ public void applyClicked(ActionEvent actionEvent) {
...
+                {
+                    contributorsField.getItems().add(user);
+                }
+            });
+
+            //Check all contributors
+            project.getContributors().forEach(user -> {
+                if(contributorsField.getItems().contains(user))
+                {
```
Comment (mdjdrn1) [inline src/main/java/textEditor/controller/EditProjectController.java]: Reformat

**Review by mdjdrn1 — CHANGES_REQUESTED** [review]
See comments

**Review by mdjdrn1 — APPROVED** [review]

**Pull request merged**